test/integration/test_keyprotect.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import unittest
+import os
+import warnings
+
+import redstone
+from redstone import auth as bxauth
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class KeyProtectTestCase(unittest.TestCase):
+
+    def setUp(self):
+        # filter out the resource warning message
+        warnings.simplefilter("ignore", ResourceWarning)
+
+        apikey = os.environ.get('IBMCLOUD_API_KEY')
+        tm = bxauth.TokenManager(api_key=apikey)
+        self.rc = redstone.service("ResourceController")
+        self.instance_id, self.crn = self.rc.create_instance(name="test-instance", plan_id=self.rc.KEYPROTECT_PLAN_ID,
+                                                             region="us-south")
+        self.kp = redstone.service("KeyProtect",
+                                   credentials=tm,
+                                   region="us-south",
+                                   service_instance_id=self.instance_id,
+                                   )
+
+    def tearDown(self):
+        for key in self.kp.keys():
+            self.kp.delete(key.get('id'))
+
+        self.rc.delete_instance(self.instance_id)
+        self.kp.session.close()
```

Comment:
You will probably want to catch the errors if one of the keys fails to delete, or if the instance fails to delete.

I would probably delete all the keys in a try block, and then in a finally clause try the delete on the instance.

Reply:
Updated, thanks!